include/h2o/memory.h

```diff
@@ -246,6 +246,14 @@ static void h2o_mem_addref_shared(void *p);
  * The chunk gets freed when the ref-count reaches zero.
  */
 static int h2o_mem_release_shared(void *p);
+
+/**
+ * creates a temporary file using to the fn_template param.
+ * This is a wrapper to mkstemp(3), but there's no way to access the actual filename.
+ * @return fd. -1 on failure and set errno.
+ */
+int h2o_make_temp_file(const char *fn_template);
+
```

Comment:
I was expecting a function that belongs to `h2o_buffer_t`, but writing a function specific for creating temporary files make sense.

Though, it we want to take that path, I think that it should belong to `include/h2o/file.h`, `lib/common/file.c`, and that the function should use `h2o_file_` as the prefix. Maybe `h2o_file_mktemp`?

Reply:
I see. `h2o_file_mktemp` in `h2o/file.h` seems the best name and place for this function. Moved a975fbd9db6854a0b8408a3b10a61f9edf333320